Add documentation for Linking API

sgny opened the pull request:
@MoOx do you think stating version requirements in the bindings would be necessary?

MoOx commented on reason-react-native/src/apis/Linking.md:
````diff
@@ -1,50 +1,112 @@
+
 ```reason
-type extra;
+openURL(string)
````
I was wondering if we should use real world snippet by including module name.
That would mean:

```reason
open ReactNative;

Linking.openURL("http://some.url")
```

We could potentially omit `open ReactNative;` as it like a requirement (at least I guess we are going to promote this usage), but we should probably show something that people can easily copy/paste/tweak without too much thing to think about.

What do you think?

sgny replied:
Sure, I'll create an example. Since it wouldn't hurt, I'll keep the open statement to make it as self-contained as possible.

sgny replied:
I revised to list the type for each method, considering that the example should be sufficient to demonstrate how the methods can be used.

Should we nevertheless have examples of how each method is called, as in the example you have given for `openURL`?

MoOx replied:
I would do that for all code blocks indeed. I found this much more clear than a "raw api" declaration in code as you can directly identify how you are going to use it.
@cknitt thoughts?

MoOx replied:
ORRR maybe we should have both like this

- code showing api, very concise & with type, not values (exactly what you did) 
- description blah blah blah
- very tiny example (like the one I shared)

And a "real world" example at the end, mixing multiple usage if possible (exactly what you did).

MoOx commented:
About version number, @cknitt think following RN version is a good idea, and since that was one of my idea, I am ok with that, so maybe we don't have to document this stuff...

sgny commented:
> About version number, @cknitt think following RN version is a good idea, and since that was one of my idea, I am ok with that, so maybe we don't have to document this stuff...

We're all in agreement on that, then. I will keep the version notes in this documentation while we decide with which version we tag our own release. That is while we decide whether we call this 0.60 or 0.5x.

cknitt commented:
I meant to request changes, not to approve, sorry!

sgny commented on reason-react-native/src/apis/Linking.md:
```diff
@@ -1,50 +1,192 @@
+open ReactNative;
+
+let windowHeight = Dimensions.get(`window)##height->float_of_int;
+let windowWidth = Dimensions.get(`window)##width->float_of_int;
```
Probably, most uses will require conversion to `float`.

sgny commented on reason-react-native/src/apis/Linking.md:
```diff
@@ -1,50 +1,192 @@
-external openURL: string => Js.Promise.t(unit) = "";
+To determine whether a URL can be opened by registered applications. URL should
+be specified as `string`, and the method returns `bool` wrapped in a promise,
+that is `Js.Promise.t(unit)`. The promise will be rejected if it is impossible
```
I was too hasty with the copy&paste; fixed.

At first I did not actually have explicit type information (as in the code blocks now), so wanted to include that in the description. With the types explicitly stated now, I don't think that explanation is necessary anywhere.

sgny commented:
> Feel free to merge, we can improve the global layout of the doc later.

I fixed the error @cknitt noticed and cleaned up the description, but individual examples might have to wait.

MoOx commented:
Thanks for all this work!
